src/org/zaproxy/zap/extension/formhandler/FormHandlerParam.java

```diff
@@ -0,0 +1,233 @@
+
+            if (field.isEnabled()) {
+                enabledFields.add(field.getName().toLowerCase());
+            }
+        }
+
+        enabledFields.trimToSize();
+        this.enabledFieldsNames = enabledFields;
+    }
+
+    /**
+     * Adds a new field with the given {@code name} and {@code value}, enabled by default.
+     * <p>
+     * The call to this method has no effect if the given {@code name} is null or empty, or a field with the given name already
+     * exist.
+     *
+     * @param name the name of the field that will be added
+     * @param value the value of the field that will be added
+     */
+    public void addField(String name, String value) {
+        if (name == null || name.isEmpty()) {
+            return;
+        }
+
+        for (Iterator<FormHandlerParamField> it = fields.iterator(); it.hasNext();) {
+            if (name.equalsIgnoreCase(it.next().getName())) {
+                return;
+            }
+        }
+
+        this.fields.add(new FormHandlerParamField(name, value));
+
+        this.enabledFieldsNames.add(name);
+    }
+
+    /**
+     * Removes the field with the given {@code name}.
+     * <p>
+     * The call to this method has no effect if the given {@code name} is null or empty, or a field with the given {@code name}
+     * does not exist.
+     *
+     * @param name the name of the field that will be removed
+     */
+    public void removeField(String name) {
+        if (name == null || name.isEmpty()) {
+            return;
+        }
+
+        for (Iterator<FormHandlerParamField> it = fields.iterator(); it.hasNext();) {
+            FormHandlerParamField field = it.next();
+            if (name.equals(field.getName())) {
+                it.remove();
+                if (field.isEnabled()) {
+                    this.enabledFieldsNames.remove(name);
+                }
+                break;
+            }
+        }
+    }
+
+    /**
+     * Gets the value for the field {@code name} by searching through all existing fields
+     *
+     * @param name the name of the field being queried.
+     * @return the value of the field
+     */
+    public String getField(String name){
+        this.setFields(fields);
+        for (FormHandlerParamField field: fields){
+            if(field.getName().equals(name)){
+                return field.getValue();
+            }
+        }
+        return null;
+    }
+
+    public List<String> getEnabledFieldsNames() {
+        return enabledFieldsNames;
+    }
+
+    /**
+     * Gets the value of the field that is enabled {@code name}.
+     * If the field exists in the current list then it will return its value
+     *
+     * @param name the name of the field that is being checked
+     * @return string of the enabled field's value
+     */
+    public String getEnabledFieldValue(String name){
+        for (String field: getEnabledFieldsNames()){
```

Comment:
I'd suggest using a private `FormHandlerParamField getParamField(String name)`, so you can obtain the enabled state and value without searching for it again.